app/controllers/api/subcollections/snapshots.rb

```diff
@@ -0,0 +1,36 @@
+module Api
+  module Subcollections
+    module Snapshots
+      def snapshots_query_resource(object)
+        object.snapshots
+      end
+
+      def snapshots_create_resource(parent, _type, _id, data)
+        validation = parent.validate_create_snapshot
+
+        if validation[:available]
+          task_id = queue_object_action(
+            parent,
+            "summary",
+            :method_name => "create_snapshot",
+            :args => [data[:name], data[:description], data[:memory]]
+          )
+          action_result(true, "Creating snapshot for #{parent.id}", :task_id => task_id)
+        else
+          action_result(false, validation[:message])
+        end
```

Comment:
also, instead of parent.id for the successful return, would be useful to have something similar to vm_ident(),  maybe something like  '#{ctype} id: #{parent.id} name: #{parent.name}'